src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1884,10 +1902,38 @@ def client_side_proxy(cmd,
     except Exception as e:
         telemetry.set_exception(exception=e, fault_type=consts.Merge_Kubeconfig_Fault_Type,
                                 summary='Unable to merge kubeconfig.')
-        raise CLIError("Failed to merge kubeconfig." + str(e))
+        if flag==0:
+            close_subprocess(clientproxy_process)
+            raise CLIError("Failed to merge kubeconfig." + str(e))
+        else:
+            thread_exception = "Failed to merge kubeconfig." + str(e)
+            stop_threads = True
+            return
+
+    # Starting a timer to refresh the credentials, 5 mins before expiry
+    fun_args = [cmd, client, resource_group_name, cluster_name, 1, args, client_proxy_port, api_server_port, operating_system, creds, user_type, debug_mode, token, path, context_name]
+    refresh_thread = Timer(expiry - time.time() - 300, client_side_proxy, args=fun_args)
+    refresh_thread.setDaemon(True)
+    try:
+        refresh_thread.start()
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Run_RefreshThread_Fault_Type,
+                                summary='Unable to run refresh thread')
+        if flag==0:
+            close_subprocess(clientproxy_process)
```

Comment:
Take both the statements into a one method close_subprocess and raise CLI_Error to avoid redundancy.

Reply:
taken care of in new approach.